bndtools.core/src/bndtools/launch/util/LaunchUtils.java

```diff
@@ -16,6 +22,16 @@
 
 public final class LaunchUtils {
 
+    private static final ILogger logger = Logger.getLogger(LaunchUtils.class);
+
+    private static final ServiceTracker<LaunchListener,LaunchListener> launchListeners;
+
+    static {
+        final BundleContext context = FrameworkUtil.getBundle(LaunchUtils.class).getBundleContext();
```

Comment:
I guess I would feel safer is `bndtools.launch.util` were not an exported package. Since it is, one day, code from another bundle could call it when bndtools.core bundle has not been started. Perhaps moving the tracker into a lazy initialization holder class would be safer to defer the initialization of the tracker until it is actually needed.

```java
private static class TrackerHolder {
    static final ServiceTracker<LaunchListener,LaunchListener> launchListeners;
    static {
        final BundleContext context = FrameworkUtil.getBundle(LaunchUtils.class).getBundleContext();
        launchListeners = new ServiceTracker<LaunchListener,LaunchListener>(context, LaunchListener.class, null);
        launchListeners.open();
    }
}

```